### ISSUE TITLE
Taint not propagating to objects with anonymous type in C#

### ISSUE THREAD

**Issue by jkielpinski** [semgrep/semgrep#7995]
**Describe the bug**
While processing C# code in taint analysis mode, Semgrep is not propagating taint into objects with anonymous type on assignment. E.g.: `var shouldBeTaintedButIsnt = new { T = TaintSource() };`

**To Reproduce**
The following Semgrep playground illustrates the issue: https://semgrep.dev/playground/s/KPXP
Semgrep matches the case where the type is not anonymous, but does not match the anonymous type case

Rule:
~~~yaml
rules:
  - id: sink_anonymous_type
    pattern-sources:
      - pattern: GetSecret()
    pattern-sinks:
      - pattern: Log(...);
    mode: taint
    message: Semgrep found a match
    languages:
      - csharp
    severity: WARNING
~~~

Test cases:
~~~csharp
class Asdf
{
    //
    // CASE 1: This case is correctly matched by Semgrep
    //
    class ReturnObject1
    {
        public string Secret;
    }
    public void Test1()
    {
        Log(new ReturnObject1() { Secret = GetSecret() });
    }

    //
    // CASE 2: This case is NOT matched
    //
    public void Test2()
    {
        Log(new { Secret = GetSecret() });
    }
}
~~~

**Expected behavior**
Semgrep would match both cases -- both anonymous and non-anonymous typed objects.

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

**Use case**
Just catching certain cases in our codebase that are slipping through

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/semgrep/issue/PA-2831/taint-not-propagating-to-objects-with-anonymous-type-in-c. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by IagoAbal** [comment]
This seems to have been fixed, presumably by PR #7975, please reopen otherwise!
